index.go

```diff
@@ -44,6 +44,15 @@ type Index interface {
 	// AddWithIDs is like Add, but stores xids instead of sequential IDs.
 	AddWithIDs(x []float32, xids []int64) error
 
+	// Applicable only to IVF indexes: Return a map of centroid ID --> []vector IDs
+	// for the cluster.
+	GetClusterAssignment() (ids map[int64][]int64, err error)
+
+	// Applicable only to IVF indexes: Returns the centroid IDs closest to the
+	// query 'x' and their distance from 'x'
+	GetCentroidDistances(x []float32, centroidIDs []int64) (
```

Comment:
Bad naming again and commentary (the closest is for the caller to determine it seems).

Consider changing to `ObtainClustersWithDistancesFromIVFIndex` or better.
(use the word cluster here if you're gonna use it in your new search API)